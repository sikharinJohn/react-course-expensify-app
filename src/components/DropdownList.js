import React from 'react';
import Select from 'react-select';

const DropdownList = (props) => (
    <Select
        options={props.types}
        value={{label:  props.selectedValue, value:  props.selectedValue }}
        onChange={props.handleChange}
        selectedValue={ props.selectedValue }
    />
);

export default DropdownList;