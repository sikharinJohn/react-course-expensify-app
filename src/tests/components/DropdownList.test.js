import React from 'react';
import { shallow } from 'enzyme';
import DropdownList from '../../components/DropdownList';
import types from '../../models/expense-types';

test ('should render DropdownList correctly', () => {
    const wrapper = shallow(<DropdownList  types={types}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should call DropdownList on change', () =>{
    const onTypeChange = jest.fn();
    const wrapper = shallow(<DropdownList 
        types={types}
        handleChange={onTypeChange} 
        />);
    wrapper.simulate('change', { target: { value: 'Gift' } });
    expect(onTypeChange).toHaveBeenCalled();
});