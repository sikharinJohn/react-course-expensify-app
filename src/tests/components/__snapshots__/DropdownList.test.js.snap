// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DropdownList correctly 1`] = `
<StateManager
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  options={
    Array [
      Object {
        "label": "Food",
        "value": "Food",
      },
      Object {
        "label": "Transport",
        "value": "Transport",
      },
      Object {
        "label": "House",
        "value": "House",
      },
      Object {
        "label": "Internet",
        "value": "Internet",
      },
      Object {
        "label": "Electric",
        "value": "Electric",
      },
      Object {
        "label": "Water",
        "value": "Water",
      },
      Object {
        "label": "Gift",
        "value": "Gift",
      },
      Object {
        "label": "Donate",
        "value": "Donate",
      },
      Object {
        "label": "Other",
        "value": "Other",
      },
    ]
  }
  value={
    Object {
      "label": undefined,
      "value": undefined,
    }
  }
/>
`;
